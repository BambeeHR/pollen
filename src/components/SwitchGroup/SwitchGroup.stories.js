import { action } from '@storybook/addon-actions';
import Form from '../../constants/Form';
import SwitchGroup from './SwitchGroup.vue';
import TypeOverline from '../TypeOverline/TypeOverline.vue';

const { Sizes } = Form;
const sizes = [Sizes.DENSE, Sizes.NORMAL];

export default {
  title: 'Toggles/SwitchGroup',
  component: SwitchGroup,
  parameters: {
    componentSubtitle: 'A group of `CheckboxInput` components',
  },
};

export const Gallery = () => ({
  components: { SwitchGroup, TypeOverline },
  data() {
    return {
      sizes,
      options: [
        'Option 1',
        {
          label: 'My Second Option',
          value: 'foo',
        },
        {
          label: 'Option 3',
          value: 'bar',
        },
      ],
      value: ['foo'],
    };
  },
  methods: {
    handleInput: action('input'),
  },
  template: `
  <div>
    <div v-for="size in sizes" :key="size">
      <TypeOverline tag="h3" class="mb-2">{{ size }}</TypeOverline>
      <SwitchGroup v-model="value" @input="handleInput" :size="size" :options="options" />
      <TypeOverline tag="h3" class="mb-2">{{ size }}, invalid</TypeOverline>
      <SwitchGroup v-model="value" @input="handleInput" :size="size" :options="options" invalid />
      <TypeOverline tag="h3" class="mb-2">{{ size }}, disabled</TypeOverline>
      <SwitchGroup v-model="value" @input="handleInput" :size="size" :options="options" disabled />
    </div>
  </div>
  `,
});

export const LabelOnTheLeft = () => ({
  components: { SwitchGroup, TypeOverline },
  data() {
    return {
      sizes,
      options: [
        'Option 1',
        {
          label: 'My Second Option',
          value: 'foo',
        },
        {
          label: 'Option 3',
          value: 'bar',
        },
      ],
      value: ['foo'],
    };
  },
  methods: {
    handleInput: action('input'),
  },
  template: `
  <div>
    <div v-for="size in sizes" :key="size">
      <TypeOverline tag="h3" class="mb-2">{{ size }}</TypeOverline>
      <SwitchGroup v-model="value" @input="handleInput" :size="size" :options="options" label-position="left" />
      <TypeOverline tag="h3" class="mb-2">{{ size }}, invalid</TypeOverline>
      <SwitchGroup v-model="value" @input="handleInput" :size="size" :options="options" label-position="left" invalid />
      <TypeOverline tag="h3" class="mb-2">{{ size }}, disabled</TypeOverline>
      <SwitchGroup v-model="value" @input="handleInput" :size="size" :options="options" label-position="left" disabled />
    </div>
  </div>
  `,
});

export const Vertical = () => ({
  components: { SwitchGroup, TypeOverline },
  data() {
    return {
      sizes,
      options: [
        'Option 1',
        {
          label: 'My Second Option',
          value: 'foo',
        },
        {
          label: 'Option 3',
          value: 'bar',
        },
      ],
      value: ['foo'],
    };
  },
  methods: {
    handleInput: action('input'),
  },
  template: `
  <div>
    <div v-for="size in sizes" :key="size">
      <TypeOverline tag="h3" class="mb-2">{{ size }}</TypeOverline>
      <SwitchGroup v-model="value" @input="handleInput" :size="size" :options="options" vertical />
      <TypeOverline tag="h3" class="mb-2">{{ size }}, invalid</TypeOverline>
      <SwitchGroup v-model="value" @input="handleInput" :size="size" :options="options" invalid vertical />
      <TypeOverline tag="h3" class="mb-2">{{ size }}, disabled</TypeOverline>
      <SwitchGroup v-model="value" @input="handleInput" :size="size" :options="options" disabled vertical />
    </div>
  </div>
  `,
});

export const VerticalWithLabelOnTheLeft = () => ({
  components: { SwitchGroup, TypeOverline },
  data() {
    return {
      sizes,
      options: [
        'Option 1',
        {
          label: 'My Second Option',
          value: 'foo',
        },
        {
          label: 'Option 3',
          value: 'bar',
        },
      ],
      value: ['foo'],
    };
  },
  methods: {
    handleInput: action('input'),
  },
  template: `
  <div>
    <div v-for="size in sizes" :key="size">
      <TypeOverline tag="h3" class="mb-2">{{ size }}</TypeOverline>
      <SwitchGroup v-model="value" @input="handleInput" :size="size" :options="options" label-position="left" vertical />
      <TypeOverline tag="h3" class="mb-2">{{ size }}, invalid</TypeOverline>
      <SwitchGroup v-model="value" @input="handleInput" :size="size" :options="options" label-position="left" invalid vertical />
      <TypeOverline tag="h3" class="mb-2">{{ size }}, disabled</TypeOverline>
      <SwitchGroup v-model="value" @input="handleInput" :size="size" :options="options" label-position="left" disabled vertical />
    </div>
  </div>
  `,
});

/* eslint-disable no-param-reassign */
[Gallery, LabelOnTheLeft, Vertical, VerticalWithLabelOnTheLeft].forEach(
  (item) => {
    item.story = {
      parameters: {
        jest: ['SwitchGroup.test.js'],
      },
    };
  }
);
