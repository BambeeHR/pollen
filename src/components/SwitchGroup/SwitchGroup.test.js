import { mount } from '@vue/test-utils';
import SwitchGroup from './SwitchGroup.vue';

const options = [
  'Option 1',
  {
    label: 'Option 2',
    value: 'foo',
  },
  {
    label: 'Option 3',
    value: 'bar',
  },
];

describe('SwitchGroup', () => {
  test('renders correctly', () => {
    const wrapper = mount(SwitchGroup, {
      propsData: {
        name: 'test',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('renders options correctly', () => {
    const wrapper = mount(SwitchGroup, {
      propsData: {
        options,
        value: ['foo'],
        name: 'test',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('emits an input event', async () => {
    const onInput = jest.fn();
    const component = {
      components: { SwitchGroup },
      data() {
        return {
          options,
          value: null,
        };
      },
      methods: {
        onInput,
      },
      template: `<SwitchGroup v-model="value" :options="options" @input="onInput" />`,
    };
    const wrapper = mount(component);
    const inputs = wrapper.findAll('input');
    expect(inputs.length).toBe(3);
    await inputs.at(1).trigger('click');
    expect(onInput).toHaveBeenLastCalledWith(['foo']);
    expect(wrapper.vm.$data.value).toEqual(['foo']);
    await inputs.at(0).trigger('click');
    expect(onInput).toHaveBeenLastCalledWith(['foo', 'Option 1']);
    expect(wrapper.vm.$data.value).toEqual(['foo', 'Option 1']);
    await inputs.at(2).trigger('click');
    expect(onInput).toHaveBeenLastCalledWith(['foo', 'Option 1', 'bar']);
    expect(wrapper.vm.$data.value).toEqual(['foo', 'Option 1', 'bar']);
    await inputs.at(2).trigger('click');
    expect(onInput).toHaveBeenLastCalledWith(['foo', 'Option 1']);
    expect(wrapper.vm.$data.value).toEqual(['foo', 'Option 1']);
  });
});
