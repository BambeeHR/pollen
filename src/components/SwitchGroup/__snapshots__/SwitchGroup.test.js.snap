// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwitchGroup renders correctly 1`] = `
<div
  class="toggle-group switch-group toggle-group--horizontal toggle-group--normal"
>
  <label
    class="toggle-input switch-input toggle-input--normal toggle-input--switch toggle-input--label-right"
    for="test-0"
    variant="standard"
  >
    <input
      class="toggle-input__input"
      id="test-0"
      name="test"
      type="checkbox"
    />
     
    <div
      class="switch-indicator switch-indicator--normal"
      variant="switch"
    />
     
    <div
      class="toggle-input__label"
    >
      <span>
        Option 1
      </span>
    </div>
  </label>
  <label
    class="toggle-input switch-input toggle-input--normal toggle-input--switch toggle-input--label-right"
    for="test-1"
    variant="standard"
  >
    <input
      class="toggle-input__input"
      id="test-1"
      name="test"
      type="checkbox"
    />
     
    <div
      class="switch-indicator switch-indicator--normal switch-indicator--checked"
      variant="switch"
    />
     
    <div
      class="toggle-input__label"
    >
      <span>
        Option 2
      </span>
    </div>
  </label>
  <label
    class="toggle-input switch-input toggle-input--normal toggle-input--switch toggle-input--label-right"
    for="test-2"
    variant="standard"
  >
    <input
      class="toggle-input__input"
      id="test-2"
      name="test"
      type="checkbox"
    />
     
    <div
      class="switch-indicator switch-indicator--normal"
      variant="switch"
    />
     
    <div
      class="toggle-input__label"
    >
      <span>
        Option 3
      </span>
    </div>
  </label>
</div>
`;
